shell/platform/tizen/accessibility_settings.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "accessibility_settings.h"
+
+#include "flutter/shell/platform/tizen/flutter_tizen_engine.h"
+#include "flutter/shell/platform/tizen/logger.h"
+
+#if defined(TV_PROFILE)
+#include <system/system_settings.h>
+#endif
+
+// SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST = 10059 has been
+// defined in system_settings_keys.h only for TV profile.
+#define SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST 10059
+
+namespace flutter {
+#if defined(TV_PROFILE)
+static void OnAccessibilityHighContrastChangedCallback(
+    system_settings_key_e key,
+    void* user_data) {
+  auto* self = reinterpret_cast<AccessibilitySettings*>(user_data);
+  self->OnAccessibilityHighContrastStateChanged();
+}
+#endif
+
+AccessibilitySettings::AccessibilitySettings(FlutterTizenEngine* engine)
+    : engine_(engine) {
+#if defined(TV_PROFILE)
+  // Add listener for accessibility high contrast.
+  system_settings_set_changed_cb(
+      system_settings_key_e(
+          SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST),
+      OnAccessibilityHighContrastChangedCallback, this);
+
+  // Set initialized value of accessibility high contrast.
+  if (engine_ != nullptr) {
+    engine_->EnableAccessibilityFeature(GetHighContrastValue());
+  }
+#endif
+}
+
+AccessibilitySettings::~AccessibilitySettings() {
+#if defined(TV_PROFILE)
+  system_settings_unset_changed_cb(system_settings_key_e(
+      SYSTEM_SETTINGS_KEY_MENU_SYSTEM_ACCESSIBILITY_HIGHCONTRAST));
+#endif
+}
+
+void AccessibilitySettings::OnAccessibilityHighContrastStateChanged() {
```

Comment:
For consistency with the `GetHighContrastValue` method,
```suggestion
void AccessibilitySettings::OnHighContrastStateChanged() {
```

Reply:
done